include/faker-cxx/internet.h

```diff
@@ -350,4 +352,77 @@ FAKER_CXX_EXPORT std::string_view domainSuffix();
  * @endcode
  */
 FAKER_CXX_EXPORT std::string anonymousUsername(unsigned maxLength);
+
+/**
+ * @brief Generates a JSON Web Token (JWT).
+ *
+ * This function generates a JWT using the provided header, payload, and reference date.
+ * If no header or payload is provided, default values will be used.
+ * The reference date is optional and can be used to set the "iat" (issued at) claim in the payload.
+ *
+ * @param header The optional header map to include in the JWT. Defaults to a standard header.
+ * @param payload The optional payload map to include in the JWT. Defaults to a standard payload.
+ * @param refDate The optional reference date to set the "iat" claim. Defaults to the current date and time.
+ *
+ * @returns A string representing the generated JWT.
+ *
+ * @code
+ * std::map<std::string, std::string> header = {{"alg", "HS256"}, {"typ", "JWT"}};
+ * std::map<std::string, std::string> payload = {{"sub", "1234567890"}, {"name", "John Doe"}, {"admin", "true"}};
+ * faker::internet::getJWTToken(header, payload); // "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9..."
+ * @endcode
+ */
+FAKER_CXX_EXPORT std::string
+getJWTToken(const std::optional<std::map<std::string, std::string>>& header = std::nullopt,
+            const std::optional<std::map<std::string, std::string>>& payload = std::nullopt,
+            const std::optional<std::string>& refDate = std::nullopt);
+
+/**
+ * @brief Returns the algorithm used for signing the JWT.
+ *
+ * This function provides the algorithm that is used to sign the JSON Web Token (JWT).
+ *
+ * @returns A string view representing the JWT signing algorithm.
+ *
+ * @code
+ * faker::internet::getJWTAlgorithm(); // "HS256"
+ * @endcode
+ */
+FAKER_CXX_EXPORT std::string_view getJWTAlgorithm();
+}
+
+namespace faker::internet::utility
+{
+/**
+ * @brief Encodes a given string to Base64 URL format.
+ *
+ * This function takes an input string and converts it into a Base64 URL encoded string.
+ * Base64 URL encoding is a variant of Base64 encoding that is URL-safe.
+ *
+ * @param input The string to be encoded.
+ *
+ * @returns A Base64 URL encoded string.
+ *
+ * @code
+ * std::string input = "Hello, World!";
+ * faker::internet::toBase64UrlEncode(input); // "SGVsbG8sIFdvcmxkIQ"
+ * @endcode
+ */
+std::string toBase64UrlEncode(const std::string& input);
+
+/**
+ * @brief Converts a map of key-value pairs to a JSON string.
+ *
+ * This function takes a map where both keys and values are strings and converts it into a JSON formatted string.
+ *
+ * @param data The map containing key-value pairs to be converted to JSON.
+ *
+ * @returns A JSON formatted string representing the input map.
+ *
+ * @code
+ * std::map<std::string, std::string> data = {{"name", "John"}, {"age", "30"}};
+ * faker::internet::toJSON(data);  // json is now "{\"name\":\"John\",\"age\":\"30\"}"
+ * @endcode
+ */
+std::string toJSON(std::map<std::string, std::string>& data);
```

Comment:
move to internet.cpp

Reply:
then tests for the util methods wouldnt work if definition wouldnt be in header

Reply:
yup, we don't need them (my previous comment was if they were in include but I think its better to keep them outside API file)

Reply:
just delete tests, and lets focus on testing main jwt function